common/ensemble_building/singlebest_ensemble.py

```diff
@@ -0,0 +1,137 @@
+import os
+from typing import List, Tuple, Union, cast
+
+import numpy as np
+
+from sklearn.pipeline import Pipeline
+
+from smac.runhistory.runhistory import RunHistory
+
+from .abstract_ensemble import AbstractEnsemble
+from ..utils.backend import Backend
+
+
+class SingleBest(AbstractEnsemble):
+    """
+    In the case of a crash, this class searches
+    for the best individual model.
+
+    Such model is returned as an ensemble of a single
+    object, to comply with the expected interface of an
+    AbstractEnsemble.
+    """
+
+    def __init__(
+        self,
+        run_history: RunHistory,
+        seed: int,
+        backend: Backend,
+    ):
+        self.seed = seed
+        self.backend = backend
+
+        # Add some default values -- at least 1 model in ensemble is assumed
+        self.indices_ = [0]
+        self.weights_ = [1.0]
+        self.run_history = run_history
```

Comment:
The reason I save it to the object is so that it is stored on it for debug. If you just pass it around and need this small dict like object, then you won't have this info.

I prefer to keep it if it is ok.